asset/src/elasticsearch_index_selector/processor.ts

```diff
@@ -0,0 +1,161 @@
+
+import {
+    DataEntity, BatchProcessor, AnyObject, set
+} from '@terascope/job-components';
+import { IndexSelectorConfig } from './interfaces';
+
+function _getWeeklyIndex(date: string) {
+    // weeks since Jan 1, 1970
+    return Math.floor(Date.parse(date) / 604800000);
+}
+const offsets = {
+    daily: 10,
+    monthly: 7,
+    yearly: 4
+};
+
+interface BulkMeta {
+    _index: string;
+    _type: string;
+    _id: string;
+    retry_on_conflict: number;
+}
+
+interface IndexSpec extends DataEntity {
+    index?: AnyObject;
+    create?: AnyObject;
+    update?: AnyObject;
+    delete?: AnyObject;
+}
+
+interface ScriptConfig {
+    file?: string;
+    source?: string;
+    params?: AnyObject;
+}
+
+interface UpdateConfig extends DataEntity {
+    upsert?: AnyObject;
+    doc?: AnyObject;
+    script?: ScriptConfig;
+}
+
+export default class IndexSelector extends BatchProcessor<IndexSelectorConfig> {
+    private formattedDate(record: DataEntity) {
+        const { date_field: dateField, timeseries } = this.opConfig;
+        let end = 10;
+
+        // date check
+        let date;
+
+        try {
+            date = new Date(record[dateField as string]).toISOString();
```

Comment:
This doesn't validate the date correctly, use the [helper function](https://github.com/terascope/teraslice/blob/master/packages/utils/src/dates.ts#L14-L23) from `@terascope/utils`.